core/src/main/java/apoc/convert/Json.java

```diff
@@ -1,21 +1,80 @@
 package apoc.convert;
 
+import apoc.meta.Meta;
 import apoc.result.MapResult;
 import apoc.util.JsonUtil;
 import apoc.util.Util;
-import org.neo4j.graphdb.Entity;
-import org.neo4j.graphdb.Node;
-import org.neo4j.graphdb.Path;
-import org.neo4j.graphdb.Relationship;
+import org.neo4j.graphdb.*;
 import org.neo4j.procedure.*;
 
 import java.io.IOException;
 import java.util.*;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+import static apoc.util.Util.labelStrings;
+import static apoc.util.Util.map;
 
 public class Json {
 
+    // visible for testing
+    public static String NODE = "node";
+    public static String RELATIONSHIP = "relationship";
+
+    private Object writeJsonResult(Object value) {
+        Meta.Types type = Meta.Types.of(value);
+        switch (type) {
+            case NODE:
+                return nodeToMap((Node) value, true);
+            case RELATIONSHIP:
+                return relToMap((Relationship) value);
+            case PATH:
+                return writeJsonResult(StreamSupport.stream(((Path)value).spliterator(),false)
+                        .map(i-> i instanceof Node ? nodeToMap((Node) i, true) : relToMap((Relationship) i))
+                        .collect(Collectors.toList()));
+            case LIST:
+                return Convert.convertToList(value).stream().map(this::writeJsonResult).collect(Collectors.toList());
+            case MAP:
+                return ((Map<String, Object>) value).entrySet()
+                        .stream()
+                        .collect(Collectors.toMap(Map.Entry::getKey,
+                                e -> writeJsonResult(e.getValue())));
+            default:
+                return value;
+        }
+    }
+
+    private Map<String,Object> relToMap(Relationship rel) {
+        Map<String, Object> mapRel = map(
+                "id", String.valueOf(rel.getId()),
+                "type", RELATIONSHIP,
+                "label", rel.getType().toString(),
+                "start", nodeToMap(rel.getStartNode(), false),
+                "end", nodeToMap(rel.getEndNode(), false));
+
+        return mapWithOptionalProps(mapRel, rel.getAllProperties());
+    }
+
+    private Map<String,Object> nodeToMap(Node node, boolean mapForNode) {
+        Map<String, Object> mapNode = map("id", String.valueOf(node.getId()));
+
+        if (mapForNode) {
```

Comment:
I put this condition only for consistency with the procedure `apoc.export.json`, which, in case of nodes, insert `"type":"node"`. Of course, without this condition the code would be cleaner. So, I can remove it without any problem if you want.

Reply:
yep, I think it would make sense to always include this piece of information. What do you think?

Reply:
Yes, it's better that way. Changed 👍